### PR TITLE
Disable the "arc land" prompt that bypasses Harbormaster build checks

### PR DESCRIPTION
**Pull request by albertywu**
Currently, "arc land" allows users to bypass blocking Harbormaster build checks via a prompt.

This is a loophole that allows users to land bad changes onto trunk.

This PR disables this loophole by immediately aborting the "arc land" operation if it detects that the Harbormaster Buildable is failing for the diff, and also:
• displays the failing blocking builds
• displays a message to the user informing them that they can BREAKGLASS in case of emergency.

---

Locally tested:

**1) Revision does not have approval**

Screenshot 2024-12-06 at 2 50 09 PM

**2) Revision is approved, but Buildable is passing (prompt is now disabled as shown in screenshot below):**

Screenshot 2024-12-06 at 2 50 27 PM

**3) Revision is approved AND Buildable is passing (land succeeds):**
Screenshot 2024-12-06 at 3 01 43 PM

### CODE REVIEW

File: src/workflow/ArcanistLandWorkflow.php [hunk 1]
```diff
@@ -1523,115 +1523,6 @@ private function restoreBranch() {
...
-    // unless the server is running code from March 2019 or newer since the
-    // API methods we need won't exist yet. We'll fall back to the older check
-    // if this one doesn't work out.
-    try {
-      $this->checkForBuildablesWithPlanBehaviors($diff_phid);
-      return;
-    } catch (ArcanistUserAbortException $abort_ex) {
-      throw $abort_ex;
-    } catch (Exception $ex) {
-      // Continue with the older approach, below.
```
Comment (albertywu) [inline src/workflow/ArcanistLandWorkflow.php:1545]: None of the code below here was actually being run, so I removed it.

I also renamed `checkForBuildables` to `ensureBuildableIsPassing` which is clearer about what the function is doing.

File: src/workflow/ArcanistLandWorkflow.php [hunk 2]
```diff
@@ -1801,9 +1801,11 @@ private function checkForBuildablesWithPlanBehaviors($diff_phid) {
         $buildable_uri);
     }
 
-    if (!phutil_console_confirm($prompt)) {
-      throw new ArcanistUserAbortException();
-    }
+    // If harbormaster builds have not succeeded, abort the workflow and inform user about BREAKGLASS for emergencies.
+    $console = PhutilConsole::getConsole();
+    $console->writeOut("\n");
+    throw new ArcanistUsageException(
```
[the remark this answers is not in the document]
Comment (albertywu) [reply src/workflow/ArcanistLandWorkflow.php]: not hard -- this is in fact what happens with this PR change (Attached screenshot to show example).

File: src/workflow/ArcanistLandWorkflow.php [hunk 3]
```diff
@@ -1665,18 +1564,17 @@ private function checkForBuildablesWithPlanBehaviors($diff_phid) {
...
 
-    if (!$raw_buildables['data']) {
+    if (!$raw_buildable['data']) {
       return;
     }
 
-    $buildables = $raw_buildables['data'];
-    $buildable_phids = ipull($buildables, 'phid');
-
+    $buildable = head($raw_buildable['data']);
```
Comment (alex-semenyuk) [inline src/workflow/ArcanistLandWorkflow.php:1571]: Do we know the case when diff_phid is associated with more then one Buildable?

**Review by alex-semenyuk — APPROVED** [review]

File: src/workflow/ArcanistLandWorkflow.php (hunk 3 above)
Comment (albertywu) [reply src/workflow/ArcanistLandWorkflow.php:1571]: No, but that should not be the case -- there should only ever be one Buildable as we rely on that for overall status of passing vs failing checks

**Pull request merged**